### PR TITLE
EOL 3.1

### PR DESCRIPTION
**Pull request by bricelam**
.NET 3.1 goes out of support on December 13 and .NET 5 is already out of support.

Creating this draft to discuss as a team what legacy information we may still want to keep in our docs.

I'm starting extremely aggressive and assuming that anyone reading our docs will be working with a supported version of EF Core (they had a year to upgrade), and that any references to previous versions and behaviors will just create distracting noise for anyone trying to understand the current, supported versions.

### TODO

- [x] Make sure the metadata dates are up to date
- [x] Figure out what to do with the platforms page

### CODE REVIEW

File: entity-framework/core/cli/powershell.md [hunk 1]
```diff
@@ -221,16 +219,16 @@ Parameters:
...
-| <nobr>`-ContextNamespace <String>`</nobr> | The namespace to use for the generated `DbContext` class. Note: overrides `-Namespace`. Added in EF Core 5.0.                                                                                                                                                           |
+| <nobr>`-Namespace <String>`</nobr>        | The namespace to use for all generated classes. Defaults to generated from the root namespace and the output directory.0.                                                                                                                           |
+| <nobr>`-ContextNamespace <String>`</nobr> | The namespace to use for the generated `DbContext` class. Note: overrides `-Namespace`.0.                                                                                                                                                           |
```
Comment (roji) [inline entity-framework/core/cli/powershell.md:223]:
```suggestion
| <nobr>`-Namespace <String>`</nobr>        | The namespace to use for all generated classes. Defaults to generated from the root namespace and the output directory.                                                                                                                             |
| <nobr>`-ContextNamespace <String>`</nobr> | The namespace to use for the generated `DbContext` class. Note: overrides `-Namespace`.                                                                                                                                                             |
```

File: entity-framework/core/extensions/index.md [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
 author: ErikEJ
-ms.date: 07/22/2022
+ms.date: 12/13/2022
 uid: core/extensions/index
 ---
 
 # EF Core Tools & Extensions
 
-These tools and extensions provide additional functionality for Entity Framework Core 3 and later.
+These tools and extensions provide additional functionality for Entity Framework Core 6 and later.
```
Comment (roji) [inline entity-framework/core/extensions/index.md:11]: Can also just drop the version, not sure it has any real value here.

File: entity-framework/core/extensions/index.md [hunk 3]
```diff
@@ -17,31 +17,25 @@ These tools and extensions provide additional functionality for Entity Framework
 
 ### LLBLGen Pro
 
-LLBLGen Pro is an entity modeling solution with support for Entity Framework and Entity Framework Core. It lets you easily define your entity model and map it to your database, using database first or model first, so you can get started writing queries right away. For EF Core: 2, 3, 5, 6.
```
Comment (roji) [inline entity-framework/core/extensions/index.md:20]: Extensions/tools may choose to support EF versions which are out-of-support - I think it's OK to continue listing those versions.

File: entity-framework/core/miscellaneous/platforms.md [hunk 4]
```diff
@@ -14,18 +16,18 @@ We want EF Core to be available to developers on all modern .NET implementations
...
-.NET Standard<sup>(1)</sup>   | 2.0        | 2.1    |
-.NET Framework<sup>(2)</sup>  | 4.7.2      |        |
-Mono                          | 5.4        | 6.4    |
-Xamarin.iOS<sup>(3)</sup>     | 10.14      | 12.16  | (use .NET 6)
-Xamarin.Mac<sup>(3)</sup>     | 3.8        | 5.16   | (use .NET 6)
-Xamarin.Android<sup>(3)</sup> | 8.0        | 10.0   | (use .NET 6)
-UWP<sup>(4)</sup>             | 10.0.16299 |        | (use [Windows App SDK](/windows/apps/windows-app-sdk/))
-Unity<sup>(5)</sup>           | 2018.1     | 2021.2 | TBD
-Tizen<sup>(6)</sup>           | 4.0        | 6.0    | (use .NET 6)
+EF Core                       | 6.0 and 7.0
```
Comment (roji) [inline entity-framework/core/miscellaneous/platforms.md:19]: There may be some value to people in leaving the older versions (e.g. I'm stuck on old X, which version of EF can I use for now, even if unsupported)

File: entity-framework/core/providers/index.md [hunk 5]
```diff
@@ -16,33 +16,27 @@ Entity Framework Core can access many different databases through plug-in librar
...
-| [EntityFrameworkCore.Jet](https://www.nuget.org/packages/EntityFrameworkCore.Jet/)                                                                                                    | Microsoft Access files          | [Bubi](https://github.com/bubibubi)                                                             |                                            | 3.1               | [readme](https://github.com/bubibubi/EntityFrameworkCore.Jet/blob/master/docs/README.md)                                                       |
-| [Teradata.EntityFrameworkCore](https://www.nuget.org/packages/Teradata.EntityFrameworkCore/)                                                                                          | Teradata Database 16.10 onwards | [Teradata](https://downloads.teradata.com/download/connectivity/net-data-provider-for-teradata) |                                            | 3.1               | [website](https://www.nuget.org/packages/Teradata.EntityFrameworkCore/)                                                                        |
```
Comment (roji) [inline entity-framework/core/providers/index.md:40]: As with tools/extensions, maybe we should leave old providers... Like someone may need to access an old SQL Server Compact database, the information may help them even if the EF version is unsupported.

**Review by roji — COMMENTED** [review]
Some small nits/comments

File: entity-framework/core/extensions/index.md (hunk 3 above)
Comment (ErikEJ) [reply entity-framework/core/extensions/index.md:20]: I recently removed extensions that only supported v 2. Maybe do the same (later) for the ones that do not support 6?

Comment (roji) [reply entity-framework/core/extensions/index.md:20]: Yeah, it's a question of how much time we want to wait... 2 is really old by now, whereas 3.1 is just now going out of support - so we may want to give it a while.

Comment (ErikEJ) [reply entity-framework/core/extensions/index.md:20]: Sure, wait until past release of 8 of course.

**Comment by bricelam** [comment]
## 📝 Design meeting notes

* Keep providers for older versions
* Keep extensions for 3.1 or newer
* Keep older versions on provider/extension compatibility lists
* Rework the platforms page
  * Get rid of the table
  * Link to .NET docs
  * Have sections for different platforms
    * Say the last version that worked with .NET Framework & MAUI
    * Add MAUI
    * Talk about AoT here?

**Comment by bricelam** [comment]
@ajcvickers @jcjiang Could you review the [Supported .NET implementations](https://review.learn.microsoft.com/en-us/ef/core/miscellaneous/platforms?branch=pr-en-us-4171) page to make sure my messaging is all in alignment?

**Review by ajcvickers — APPROVED** [review]
Looks good.

**Pull request merged**